kafka-rest/src/main/java/io/confluent/kafkarest/resources/ConsumerGroupsResource.java

```diff
@@ -0,0 +1,139 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.kafkarest.resources;
+
+import io.confluent.kafkarest.KafkaRestContext;
+import io.confluent.kafkarest.Versions;
+import io.confluent.kafkarest.entities.ConsumerEntity;
+import io.confluent.kafkarest.entities.ConsumerGroup;
+import io.confluent.kafkarest.entities.TopicName;
+import io.confluent.rest.annotations.PerformanceMetric;
+import scala.Option;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Provides metadata about consumers groups
+ */
+@Path("/groups")
+@Produces({Versions.KAFKA_V1_JSON_WEIGHTED, Versions.KAFKA_DEFAULT_JSON_WEIGHTED,
+        Versions.JSON_WEIGHTED, Versions.KAFKA_V2_JSON_WEIGHTED})
+@Consumes({Versions.KAFKA_V1_JSON, Versions.KAFKA_DEFAULT_JSON, Versions.JSON,
+        Versions.GENERIC_REQUEST, Versions.KAFKA_V2_JSON})
+public class ConsumerGroupsResource {
+
+  private final KafkaRestContext context;
+
+  /**
+   * <p>Create consumers group resource</p>
+   *
+   * @param context - context of rest application
+   */
+  public ConsumerGroupsResource(KafkaRestContext context) {
+    this.context = context;
+  }
+
+  /**
+   * <p>Get consumer group list</p>
+   * <p>Example: http://127.0.0.1:2081/groups/</p>
+   *
+   * @param count - Optional parameter. Use for paging.
+   *             Restrict count of returned entities with group information.
+   * @param offset - Optional parameter. Use for paging.
+   *               Offset which starts return records from.
+   * @return List[ConsumerGroup] -
+   *     [{"groupId":"testGroup", "coordinator": {"host": "127.0.0.1", "port": "123"}}]
+   */
+  @GET
+  @PerformanceMetric("groups.list")
+  public List<ConsumerGroup> list(@QueryParam("offset") Integer offset,
+                                  @QueryParam("count") Integer count)
+          throws Exception {
+    return context.getGroupMetadataObserver()
+            .getConsumerGroupList(Option.apply(offset), Option.apply(count));
+  }
+
+  /**
+   * <p>Get partitions list for group groupId</p>
+   * <p>Example: http://127.0.0.1:2081/groups/testGroup/partitions</p>
+   *
+   * @param groupId - Group name.
+   * @return ConsumerEntity
+   */
+  @GET
+  @Path("/{groupId}/partitions")
+  @PerformanceMetric("groups.get.partitions")
+  public ConsumerEntity getPartitionsInformation(@PathParam("groupId") String groupId)
```

Comment:
What is the reason for choosing to duplicate the data shown in `ConsumerEntity`? I am specifically asking about the repeating `consumerId` and `host` info in each TopicPartitionEntity. It seems cleaner if we had a `consumer` object that had a list of TopicPartitions mapped to their offsets/lag.

Reply:
To my mind we may need to request information about which host who is the member of specified group. I've included this information into TopicPartitionEntity(now renamed) because each partition is able to be read by different consumer.

Reply:
Yes but in the common case that's not the case, right? We could have something like:
```
{ partition: 0, consumerId: 1, host: "localhost:9092" },
{ partition: 1, consumerId: 1, host: "localhost:9092" },
{ partition: 2, consumerId: 1, host: "localhost:9092" },
```
Most typical Kafka users don't run big consumer groups. I think it would be cleaner if we had the following format:
```
[
  consumer: { consumerId: 1, host: "localhost:9092" },
  partitions: [0, 1, 2]
]
```
What are your thoughts?

Reply:
Okay. I suggest removing that information from model. But in the next PR I suggest returning that information by endpoint - /groups/{groupId} - 
`
[
  { consumerId: "<consumer_id>", host: "<consumer_ip>", subscription-partitions: [0, 1, 2] },
...
]
`

Do you agree to my offer?

Reply:
That sounds good to me.

What would `/groupId/partitions` return now? Just the partitions? Sounds fair if so

Reply:
@stanislavkozlovski, we may return information about group current offsets, lag, end-offsets only. Do you agree with this case?